crates/conformance-test-utils/src/madara.rs

```diff
@@ -274,4 +326,130 @@ mod tests {
+        // Given
+        let test_environment = Arc::new(KakarotTestEnvironment::new().await);
+
+        // When
+        // Use genesis_set_storage_kakarot_contract_account define the storage data
+        // to be stored into the contract account
+        let genesis_address = FieldElement::from_str("0x1234").unwrap();
+        let expected_key =
+            U256::from_str("0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb").unwrap();
+        let expected_value =
+            U256::from_str("0xccccccccccccccccccccccccccccccccdddddddddddddddddddddddddddddddd").unwrap();
+        let genesis_storage_data =
+            genesis_set_storage_kakarot_contract_account(genesis_address, expected_key, expected_value).unwrap();
+
+        // Create an atomic reference to the test environment to avoid dropping it
+        let env = Arc::clone(&test_environment);
+        // It is not possible to block the async test task, so we need to spawn a blocking task
+        tokio::task::spawn_blocking(move || {
+            // Get lock on the Starknet sequencer
+            let mut starknet = env.sequencer().sequencer.starknet.blocking_write();
+            let mut storage = HashMap::new();
+
+            // Prepare the record to be inserted into the storage
+            genesis_storage_data.into_iter().for_each(|((_, k), v)| {
+                let storage_key = StarknetStorageKey(Into::<StarkFelt>::into(k.0).try_into().unwrap());
+                let storage_value = Into::<StarkFelt>::into(v.0);
+                storage.insert(storage_key, storage_value);
+            });
+
+            // Set the storage record for the contract
+            let contract_account_class_hash = env.kakarot().contract_account_class_hash;
+            let genesis_address = StarknetContractAddress(Into::<StarkFelt>::into(genesis_address).try_into().unwrap());
+            let storage_record = StorageRecord {
+                nonce: Nonce(StarkFelt::from(0u8)),
+                class_hash: ClassHash(contract_account_class_hash.into()),
+                storage,
+            };
+            starknet.state.storage.insert(genesis_address, storage_record);
+        })
+        .await
+        .unwrap();
+
+        // Deploy the contract account with the set genesis storage and retrieve the storage on the contract
+        let starknet_client = test_environment.client().starknet_provider();
+        let genesis_contract = ContractAccount::new(&starknet_client, genesis_address);
+        let [key_low, key_high] = split_u256_into_field_elements(expected_key);
+        let actual_value =
+            genesis_contract.storage(&key_low, &key_high, &StarknetBlockId::Tag(BlockTag::Latest)).await.unwrap();
+
+        // Assert that the value stored in the contract is the same as the value we set in the genesis
+        assert_eq!(expected_value, actual_value);
+    }
+
+    #[test]
+    fn test_split_u256_into_field_elements() {
+        let test_cases = vec![
```

Comment:
this is ok for this PR but generally speaking you **don't** want to loop over test cases **inside** the test but rather to _parametrize_ the test because you indeed have 3 tests here and not one: you don't want to have all the tests failing when one case is failing. In python, we used `@pytest.mark.parametrize`, in rust it seems like we could try to use https://github.com/la10736/rstest, would you be interested in trying to add this to the project and do some test refactoring?

Reply:
Nice was actually thinking to explore some how to have fixtures and parametrize tests in Rust, I'd be keen to include it in the project! Let me create an issue 💪